pkg/sysregistriesv2/system_registries_v2.go

```diff
@@ -0,0 +1,199 @@
+package sysregistriesv2
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/url"
+	"path/filepath"
+	"strings"
+
+	"github.com/BurntSushi/toml"
+	"github.com/containers/image/types"
+)
+
+// systemRegistriesConfPath is the path to the system-wide registry
+// configuration file and is used to add/subtract potential registries for
+// obtaining images.  You can override this at build time with
+// -ldflags '-X github.com/containers/image/sysregistries.systemRegistriesConfPath=$your_path'
+var systemRegistriesConfPath = builtinRegistriesConfPath
+
+// builtinRegistriesConfPath is the path to the registry configuration file.
+// DO NOT change this, instead see systemRegistriesConfPath above.
+const builtinRegistriesConfPath = "/etc/containers/registries.conf"
+
+// tomlURL is an abstraction required to unmarshal the non-primitive and
+// non-local url.URL type when loading the toml config.
+type tomlURL struct {
+	url url.URL
+}
+
+// Mirror represents a mirror. Mirrors can be used as pull-through caches for
+// registries.
+type Mirror struct {
+	// The mirror's URL
+	URL tomlURL `toml:"url"`
```

Comment:
This use of `tomlURL` is still visible to consumers (in that they have to say `Mirror.URL.url` - and is the private `url` field even visible?

I was thinking something like
```go
// All existing docstrings [this is the API provided to Golang callers]
type Mirror struct {
    URL url.URL
    Insecure bool
}

// [this is the raw format of the config file]
type tomlMirror struct {
    URL tomlURL
    Insecure bool
}
```
unmarshal into `tomlMirror`, and then initialize a `Mirror` from it. (Maybe the `toml` package supports a nicer way to achieve the same thing.)

Reply:
I will look into this.